modules/container-service/managed-clusters/main.bicep

```diff
@@ -619,6 +631,20 @@ module managedCluster_roleAssignments '.bicep/nested_roleAssignments.bicep' = [f
   }
 }]
 
+resource dnsZone 'Microsoft.Network/dnsZones@2018-05-01' existing = if (dnsZoneResourceId != null && webApplicationRoutingEnabled) {
+  name: last(split(dnsZoneResourceId, '/'))!
+}
+
+resource dnsZone_roleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = if (dnsZoneResourceId != null && webApplicationRoutingEnabled) {
```

Comment:
Hey @samuelya, I wonder if it would make sense to add an additional condition (true/false - default: true) to control the role assignment with. Imagining a case where somebody wants to provide the DNS Zone but where the deploying SPN doesn't have `User Access Administrator` permissions I could imaging this could be a pitfall. 
I guess it is safe to say that is a general 'issue' to consider for CARML as we have other cases where we assign roles too. @eriqua any thoughts from your side too?

Reply:
PS: I am in favor of the role assignment and just want to make sure we do it the right way ;)